### PR TITLE
Removed access token from RouteOptions

### PR DESCRIPTION
**Pull request by LukasPaczos**
Closes https://github.com/mapbox/mapbox-java/issues/1275. Moves the `accessToken` parameter from `RouteOptions` to `MapboxDirections`.

### CODE REVIEW

**Comment by LukasPaczos** [comment]
Integration of this change can be seen in https://github.com/mapbox/mapbox-navigation-android/pull/4670.

File: services-directions/src/test/java/com/mapbox/api/directions/v5/MapboxDirectionsTest.java [hunk 1]
```diff
@@ -544,14 +589,15 @@ public void callForUrlLength_longUrl() {
...
       .routeOptions(
         RouteOptions.builder()
           .baseUrl(mockUrl.toString())
           .coordinatesList(fakeCoordinates)
           .profile(DirectionsCriteria.PROFILE_DRIVING_TRAFFIC)
-          .accessToken("token")
           .build()
-      );
+      )
+      .accessToken("token");
```
Comment (korshaknn) [inline services-directions/src/test/java/com/mapbox/api/directions/v5/MapboxDirectionsTest.java:600]: token set twice

File: services-directions/src/test/java/com/mapbox/api/directions/v5/MapboxDirectionsTest.java [hunk 2]
```diff
@@ -669,13 +723,21 @@ public void depart_at() throws Exception {
...
+      .accessToken("token")
       .routeOptions(routeOptions.toBuilder().baseUrl(mockUrl.toString()).build())
       .build();
 
     assertEquals("true",
       mapboxDirections.cloneCall().request().url().queryParameter("enable_refresh"));
   }
 
+  @Test(expected = Exception.class)
+  public void token_required() throws Exception {
```
Comment (korshaknn) [inline services-directions/src/test/java/com/mapbox/api/directions/v5/MapboxDirectionsTest.java:735]: do we provide any readable message?
can we test it?

Comment (LukasPaczos) [reply services-directions/src/test/java/com/mapbox/api/directions/v5/MapboxDirectionsTest.java:735]: We can't this is implemented directly in the `AutoValue` library - if there's a non-nullable field that wasn't initialized, `AutoValue` will throw an exception saying which field is missing. I added the test mostly for us to remember to keep the `accessToken` as a `non-nullable field in the future.

**Review by RingerJK — APPROVED** [review]
LGTM

**Review by korshaknn — APPROVED** [review]

**Pull request merged**